stream-chat-android-client/src/main/java/io/getstream/chat/android/client/StreamLifecycleObserver.kt

```diff
@@ -4,7 +4,7 @@ import androidx.lifecycle.Lifecycle
 import androidx.lifecycle.LifecycleObserver
 import androidx.lifecycle.OnLifecycleEvent
 import androidx.lifecycle.ProcessLifecycleOwner
-import io.getstream.chat.core.internal.coroutines.DispatcherProvider
+import io.getstream.chat.android.core.internal.coroutines.DispatcherProvider
```

Comment:
Do we need to move it into `android`? :thinking: 
What about if in the future we release a "LLC pure Kotlin"? It could use this `DispatcherProvider` but doesn't have any "android related stuff"

Reply:
In that case, the `Main` dispatcher wouldn't make any sense, so this is definitely Android specific.

Even if we have things right now that are pure though, I think we can should keep them in the Android packages. Until we're explicitly working towards a pure Kotlin SDK, we're not paying attention to designing for it anyway, so we'd be bound to make incorrect assumptions along the way, putting things into the pure module that are actually platform specific somehow.

Reply:
But right now it's not android specific, is it? Main Dispatcher has meaning not only in android world
![image](https://user-images.githubusercontent.com/11807573/99786347-6ac9cf00-2b2f-11eb-90fa-83f176134153.png)

Reply:
Yes, technically, the Main dispatcher can map to different things, but I don't see us actually using it for any of those in practice, at least not the same way as we're using it now on Android.

Goes back to the previous point, if we haven't designed for it and figured out how it would work exactly, it's better not to have a pure module for now, because if anything, it'll be misleading to have it and think it's easy to reuse outside of Android.

Reply:
For now, `stream-chat-core` is pure kotlin, no Android stuff related.
The `Dispatcher.Main` is not Android related, you need to indicated which "main dispatcher" you want to use depending on which platform you are

Reply:
The module itself is called `stream-chat-android-core`, it's a core module for Android related things. It's pure Kotlin by coincidence. If we want a pure core module, that really should be named `stream-chat-core` (or `stream-chat-jvm-core`, perhaps). We can add that later if it's actually necessary and useful. For the time being, we have no real need for it.

Reply:
Ok, out-of-scope of this PR. We will rename this module in another PR